pravega-client-examples/src/main/java/io/pravega/example/streamprocessing/AppConfiguration.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ */
+package io.pravega.example.streamprocessing;
+
+import java.net.URI;
+import java.util.UUID;
+
+/**
+ * All parameters will come from environment variables.
+ * This makes it easy to configure on Docker, Kubernetes, etc.
+ */
+class AppConfiguration {
+    AppConfiguration(String[] args) {
+    }
+
+    // By default, we will connect to a standalone Pravega running on localhost.
+    public URI getControllerURI() {
+        return URI.create(getEnvVar("PRAVEGA_CONTROLLER", "tcp://localhost:9090"));
+    }
+
+    public String getScope() {
+        return getEnvVar("PRAVEGA_SCOPE", "examples");
+    }
+
+    public String getInstanceId() {
+        return getEnvVar("INSTANCE_ID", UUID.randomUUID().toString());
+    }
+
+    public String getStream1Name() {
```

Comment:
AppConfiguration is shared by EventGenerator, AtLeastOnceApp, and EventDebugSink. Stream 1 is the output for EventGenerator and the input for AtLastOnceApp. Stream 2 is the output for AtLeastOnceApp and the input for EventDebugSink. To avoid the confusion from this point of view, I chose Stream 1 and Stream 2. It is a little odd, I admit.